### ISSUE TITLE
Analyse the difference between names held in DQT and in the ECF1 database

### ISSUE THREAD

**Issue by claire-hughez** [DFE-Digital/ecf2#179]
## Why?
We want to find out how different the names are that are held in the DQT and ECF1 database.

We know that the first name will have to be the same (unless edited later) because of the checks when an ECT is added.

However, the surname could be different. We want to know how common this is. This is because we want to try copying the Claim funding for mentors service and rely on using only TRN and DOB (and National insurance number when necessary) to find a record, and then use the name saved in the DQT. This means for the name to change, it would have to be changed in DQT. 

There's evidence that DQT mismatches are causing a lot of support tickets and this might help.

We think this is more likely to be for mentors - as they'll have older records on average - so the analysis could focus on just mentors. If easy, it might be helpful to look at a sample of ECTs as well.

## What?
- Get TRN and name data for mentors/ECTs  from ECF1 database
- Get TRN and name data from DQT database, that matches the data here
- Assess how often the name is different 
- Could we also see how email differs?

## Out of scope
I don't think we need to worry about many other fields, but happy to be challenged on what else might be interesting to compare or consider for this analysis. 

## Helpful links or other information
Feedback from support on the DQT mismatch issue: https://lucid.app/lucidspark/03e5c392-88ab-4ea0-a419-c2dbb0b4b5a2/edit?invitationId=inv_181be65e-419b-456e-90f2-fddc1c00e0f0&page=0_0#
Design history for Claim funding for mentor training on how they validate records with DQT: https://becoming-a-teacher.design-history.education.gov.uk/claim-funding-for-mentors/adding-date-of-birth-to-the-add-mentor-flow/

**Comment by claire-hughez** [comment]
Devs might be able to help with CPD, Nathan might be able to get DQT data.

We think this might be quite tricky to get the data, so let's ask around.

**Comment by mason-emily** [comment]
From chatting to @tonyheadford we might be able to quite quickly script something to do a random subset of people to compare names via our API calls

**Comment by tonyheadford** [comment]
I have performed some randomised queries against mentors in ECF given the thought that as they've been in teaching longer there's more chance for the name to drift from DQT. tldr; about 80% correct.

For the comparison I converted both the ECF `full_name` and the DQT `lastName` fields to lower case and compared them by checking if the ECF `full_name` ended with the DQT lastName.

I ran 6 batches of random samples of 100 and averaged the result of how many were correct and it was 80%:
(matching result numbers were 78, 80, 77, 88, 77 and 82).

I then ran 2 larger random sample of 1000 and got these numbers 802 and 793 matching 79.7%
